src/features/post/new/PostEditorRoot.tsx

```diff
@@ -419,6 +421,26 @@ export default function PostEditorRoot({
                   value={url}
                   onIonInput={(e) => setUrl(e.detail.value ?? "")}
                 />
+                {!!jwt && !!url && isValidUrl(url) && (
+                  <IonButton
+                    onClick={(e) => {
+                      e.preventDefault();
+                      function toast() {
+                        presentToast({
+                          message: "Unable to fetch title",
+                          color: "danger",
+                        });
+                      }
+                      getSiteMetadata(url, instanceUrl, jwt)
```

Comment:
instead of adding this function to lemmy.ts, add `const client = useClient()` to the top of the component and then call `client.getSiteMetadata()` directly (`const { metadata } = await client.getSiteMetadata()`)

something like-

```ts
let metadata;

try {
  ({ metadata } = await client.getSiteMetadata())
} catch(error) {
  presentToast(siteMetadataError)
  throw error;
}

if (!data.title) {
  presentToast(siteMetadataError)
  return
}

setTitle(metadata.title)
```